priv/templates/phx.gen.auth/auth.ex

```diff
@@ -107,6 +108,24 @@ defmodule <%= inspect auth_module %> do
     end
   end
 
+  @doc """
+  Authenticates and mounts in the socket assigns the
+  <%= schema.singular %> by looking into the session.
+  """
+  def mount_current_<%= schema.singular %>(session, socket) do
```

Comment:
we can make this private. Let's also add an `on_mount(:ensure_authenticated, ...` clause which enforces authentication. It can call the shared private `mount_current_user` function